internal/config.go

```diff
@@ -2,45 +2,45 @@ package internal
 
 import (
 	"encoding/json"
-	"github.com/go-yaml/yaml"
-	"github.com/wal-g/wal-g/internal/tracelog"
 	"io/ioutil"
 	"os"
 	"os/user"
 	"path/filepath"
 	"strings"
+
+	"github.com/go-yaml/yaml"
+	"github.com/wal-g/wal-g/internal/tracelog"
 )
 
 var (
 	WalgConfig        *map[string]string
 	allowedConfigKeys = map[string]*string{
-		"WALG_S3_PREFIX":                 nil,
-		"WALE_S3_PREFIX":                 nil,
-		"WALG_FILE_PREFIX":               nil,
-		"WALE_FILE_PREFIX":               nil,
-		"WALG_GS_PREFIX":                 nil,
-		"WALE_GS_PREFIX":                 nil,
-		"GOOGLE_APPLICATION_CREDENTIALS": nil,
```

Comment:
Currently, the user can set up this var using not just env var, but also with yaml or json config of WAL-G. If we rely only on GCP SDK this will not work. How do you think, is this functionality useful, or we can simply drop it? Let's also ask @carolfly86 and @NikolayS on their opinion, since they were asking about GCP.

Reply:
Envvar + metadata server is the "standard" way to give credentials to an app using a GCP client library. The S3 integration does the same, and wal-e's GS and S3 too.

I agree with you that it's somewhat inconsistent though... We can also get `GOOGLE_APPLICATION_CREDENTIALS` via wal-g's config system, pass it to the GCS client lib if set, and if not let the lib do the default (which will check `GOOGLE_APPLICATION_CREDENTIALS` again, and fall back to metadata server). 

But then we should do the same for `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY,` `AWS_SECURITY_TOKEN`, `AWS_PROFILE` ... for consistency.

Reply:
And we probably don't want to do that because it gets way more complex than google's :)